back/spec/services/surveys/results_generator_spec.rb

```diff
@@ -0,0 +1,700 @@
+              2 => { title_multiloc: { 'en' => '2 - Disagree', 'fr-FR' => '2 - Être en désaccord', 'nl-NL' => '2 - Niet mee eens' } },
+              3 => { title_multiloc: { 'en' => '3 - Slightly disagree', 'fr-FR' => '3 - Plutôt en désaccord', 'nl-NL' => '3 - Enigszins oneens' } },
+              4 => { title_multiloc: { 'en' => '4 - Neutral', 'fr-FR' => '4 - Neutre', 'nl-NL' => '4 - Neutraal' } },
+              5 => { title_multiloc: { 'en' => '5 - Slightly agree', 'fr-FR' => "5 - Plutôt d'accord", 'nl-NL' => '5 - Enigszins eens' } },
+              6 => { title_multiloc: { 'en' => '6 - Agree', 'fr-FR' => "6 - D'accord", 'nl-NL' => '6 - Mee eens' } },
+              7 => { title_multiloc: { 'en' => '7 - Strongly agree', 'fr-FR' => "7 - Tout à fait d'accord", 'nl-NL' => '7 - Strerk mee eens' } }
+            }
+          }
+        }
+      end
+
+      it 'returns the results for a linear scale field' do
+        expected_result_linear_scale[:questionNumber] = 4
+        expect(generated_results[:results][4]).to match expected_result_linear_scale
+      end
+
+      it 'returns a single result for a linear scale field' do
+        expect(generator.generate_result_for_field(linear_scale_field.id)).to match expected_result_linear_scale
+      end
+
+      context 'when not all minimum and maximum labels are configured for linear scale fields' do
+        let(:expected_result_linear_scale_without_min_and_max_labels) do
+          expected_result_linear_scale.tap do |result|
+            result[:multilocs][:answer][1][:title_multiloc] = {
+              'en' => '1',
+              'fr-FR' => "1 - Pas du tout d'accord",
+              'nl-NL' => '1'
+            }
+            result[:multilocs][:answer][5][:title_multiloc] = {
+              'en' => '5 - Slightly agree',
+              'fr-FR' => '5',
+              'nl-NL' => '5'
+            }
+          end
+        end
+
+        before do
+          linear_scale_field.update!(
+            linear_scale_label_1_multiloc: { 'fr-FR' => "Pas du tout d'accord" },
+            linear_scale_label_5_multiloc: { 'en' => 'Slightly agree' }
+          )
+        end
+
+        it 'returns minimum and maximum labels as numbers' do
+          expect(generator.generate_result_for_field(linear_scale_field.id)).to match expected_result_linear_scale_without_min_and_max_labels
+        end
+      end
+    end
+
+    describe 'rating field' do
+      let(:expected_result_rating) do
+        {
+          customFieldId: rating_field.id,
+          inputType: 'rating',
+          question: {
+            'en' => 'How satisfied are you with our service?',
+            'fr-FR' => 'À quel point êtes-vous satisfait de notre service ?',
+            'nl-NL' => 'Hoe tevreden ben je met onze service?'
+          },
+          required: true,
+          grouped: false,
+          description: { 'en' => 'Please rate your experience from 1 (poor) to 5 (excellent).' },
```

Comment:
It seems the answers cover  a scale of 1 to 7. Is this mismatch with the description intended?